src/lib/Behaviors/ResizeColumnBehavior.tsx

```diff
@@ -127,4 +147,85 @@ export class ResizeColumnBehavior extends Behavior {
     }
     return offset;
   }
+
+  handleDoubleClick(event: PointerEvent, location: PointerLocation, state: State): State {
+    this.initialLocation = location;
+    this.resizedColumn = location.column;
+    // If the cell in the current column contains a custom cell template, the width is not adjusted because the width of the cell cannot be predicted
+    if (this.findCustomCell(this.resizedColumn.idx, state.cellMatrix.rows)) {
+      return { ...state, linePosition: -1 };
+    }
+    const newWidth = this.calculateOptimalColumnWidth(this.resizedColumn.idx, state);
+    if (state.props?.onColumnResized) {
+      const newColWidth =
+        newWidth >= (state.props?.minColumnWidth ?? CellMatrix.MIN_COLUMN_WIDTH)
+          ? newWidth
+          : state.props?.minColumnWidth ?? CellMatrix.MIN_COLUMN_WIDTH;
+      state.props.onColumnResized(this.resizedColumn.columnId, newColWidth, state.selectedIds);
+    }
+    let focusedLocation = state.focusedLocation;
+    if (focusedLocation !== undefined && this.resizedColumn.columnId === focusedLocation.column.idx) {
+      const column = { ...focusedLocation.column, width: newWidth };
+      focusedLocation = { ...focusedLocation, column };
+    }
+    return { ...state, linePosition: -1, focusedLocation };
+  }
+
+  /**
+   * Iterate through all the cells in the column, finding the widest content, and return the new column width.
+   * @param idx The column index
+   * @param state The state
+   * @returns
+   */
+  calculateOptimalColumnWidth(idx: number, state: State): number {
+    let maxWidth = 0;
+
+    state.cellMatrix.rows.forEach((row) => {
+      const cell = row.cells[idx];
+      const content = this.getContentFromCell(cell, state.cellTemplates);
+      const contentWidth = this.measureTextWidth(content);
+      // The chevron cell contains arrows
+      const additionalWidth = cell.type === "chevron" ? 10 : 0;
+
+      maxWidth = Math.max(maxWidth, contentWidth + additionalWidth);
+    });
+
+    // Add some extra space in case the text gets truncated
+    return maxWidth + 20;
+  }
```

Comment:
It looks good, an alternative idea would be to create a hidden div element and set its content to the cell content. However, using a canvas seems like a more sensible idea, as it doesn't require any DOM manipulation. I think it is also worth considering creating an additional field here, which would reduce the number of calculations in case of repeating content in a given column. In this way:

```
private widthCache: { [key: string]: number } = {};

calculateOptimalColumnWidth(idx: number, state: State): number {
    let maxWidth = 0;

    state.cellMatrix.rows.forEach((row) => {
        const cell = row.cells[idx];
        const content = this.getContentFromCell(cell, state.cellTemplates);

        // Check if the width of this content is in the cache
        if (this.widthCache[content] !== undefined) {
            maxWidth = Math.max(maxWidth, this.widthCache[content]);
        } else {
            // If not, measure it and store it in the cache
            const contentWidth = this.measureTextWidth(content);
            this.widthCache[content] = contentWidth;
            maxWidth = Math.max(maxWidth, contentWidth);
        }
    });

    // Add some extra space in case the text gets truncated
    return maxWidth + 20;
}
```

Reply:
@webloopbox Yes, using a canvas can improve performance.
widthCache is really necessary to add, to reduce duplicate content calculations, which can help improve performance

Reply:
done